New/landing page

alexandrtovmach opened the pull request:
<!--
Please read the [Code of Conduct](https://github.com/nodejs/nodejs.dev/blob/master/CODE_OF_CONDUCT.md) and the [Contributing Guidelines](https://github.com/nodejs/nodejs.dev/blob/master/CONTRIBUTING.md) before opening a pull request.
-->

## Description

Started working on landing page implementation. Please review it, because any feedback will be useful.

- using scss modules for styling
- refactored file structure

https://www.loom.com/share/68d6186bfd3c4abcabe80e2df81249c1

alexandrtovmach commented:
Finally...

I've merged changes from #325 with my changes and filestructure. Refactored code a bit and I think basic version for landing page is done

@amiller-gh @ogonzal87 Please take a look

alexandrtovmach commented:
/gcbrun

alexandrtovmach commented:
/gcbrun

alexandrtovmach commented:
/gcbrun

alexandrtovmach commented:
@nodejs/website-redesign please review it)

ahmadawais commented:
/gcbrun

ogonzal87 commented:
Thank you @alexandrtovmach! This is great. Hopefully, the refactor wasn't that bad. Minor comments added. 👍

bnb commented on test/pages/__snapshots__/index.test.tsx.snap:
```diff
@@ -0,0 +1,90 @@
+  <CommunitySection
+    content={
+      Object {
+        "desc": "We’ll never share your information and always respect your inbox - quality content only, we promise.",
```
I love this assertion. I'd like to get a strict definition from the Foundation of what this means – what _their_ usage of the list is – so we can actually detail how we define this when someone will inevitably ask.

alexandrtovmach replied:
same reason as here https://github.com/nodejs/nodejs.dev/pull/324#discussion_r331869223

alexandrtovmach commented on test/pages/__snapshots__/index.test.tsx.snap:
```diff
@@ -0,0 +1,90 @@
+      }
+    }
+  />
+  <PartnerSection
```
This content is going to be merged only to `staging`, so no validations needed I think. All content added in replacable GraphQL manner, so we can easily update it later.

alexandrtovmach commented on src/components/LandingPage/Sections/SandboxSection/index.tsx:
```diff
@@ -0,0 +1,25 @@
+  <section className={styles.sandboxSection}>
+    <div className={styles.sandbox} />
+    <div className={styles.flyingAnimation}>
+      <span></span>
```
These illustrations are implemented by pure CSS, so yeah it looks a bit different, but It's better from code perspective. Of course, we need to make it more stylish, but only with CSS I think 

P.S. this illustration generates randomly every build

alexandrtovmach commented on src/components/Button/button.module.scss:
```diff
@@ -0,0 +1,32 @@
+  font-weight: var(--font-weight-semibold);
+  padding: 0;
+  margin: 0;
+  border-radius: 5.6rem;
```
I actually not very clear with the UX points like "why do we need to have a two different types of buttons on website?", but I'll add it

alexandrtovmach commented:
/gcbrun

alexandrtovmach commented:
@nodejs/website-redesign please review it)

amiller-gh commented:
Sorry for the delay @alexandrtovmach! Will take a look after the meeting today 👍

ogonzal87 commented:
Same here! If not today, I'll take a look at it tomorrow morning. 👍🏽

alexandrtovmach commented:
/gcbrun

MylesBorins commented:
@ahmadawais if the PR comes from a fork github will not trigger the link being posted. It will always follow the same pattern though

https://storage.googleapis.com/staging.nodejs.dev/$SHA/index.html

/gcbrun

https://storage.googleapis.com/staging.nodejs.dev/c5006ed/index.html

alexandrtovmach commented:
@nodejs/website-redesign ping

Let's done with that and continue working with this project structure

ogonzal87 commented:
Hey @alexandrtovmach Sorry I took so long to take a look at your pull requests! 😞  It has been crazy the last couple of weeks. I wanted to address two things before merging your PR.

1. I am not sure if we ever shared this with you but is there any way we could match the styles to the specs in the mock/prototype.
  - [Specs](https://www.figma.com/file/lOxAGGg5KXb6nwie7zXkz6/NJ-Design-System?node-id=1338%3A13127).
  - [Prototype in Figma](https://www.figma.com/proto/lOxAGGg5KXb6nwie7zXkz6/NJ-Design-System?node-id=1338%3A13127&viewport=795%2C120%2C0.10941388458013535&scaling=min-zoom).
 - [Video of me trying to explain how to loo for the styling attributes in Figma](https://www.loom.com/share/eae4b1c25f9b4552adc4fadb0e1c20f8)

2. We discussed today the illustrations in css vs SVG. We think that the best approach for now is to use the SVG's. I ran a parformance test and the hit is not big enough for the ROI we get by dedicating the time to make it look 💯.

Let me know if you have any questions. 👍 and thank you so much for your patience!

alexandrtovmach commented:
@ogonzal87 Thank you for review, and for detailed explanation in video. Appreciate it!

Here're next changes:
- replaced CSS leafs with SVG and added simple animation on hover
	<details>
		<summary>Example video</summary>
		https://www.loom.com/share/70a6b15c9aed4e4ab38d2a2202bafd10
	</details>
- moved all design tokens to separate file `src/styles/tokens.scss`
- updated background hexagon style to fit spec

But I want to discuss point about "match the styles to the specs". I'm not sure what exactly did you mean by that, and I just want to explain something.

On design specs we have absolute CSS styles for position and size, but its horrible idea to use it on real project. The Best Practice for development is to have *reusable*, *relative* and *flexible* styles. 

<details>
	<summary>Instead of writing styles for each element...</summary>

```css
#demo-button {
	width: 115px;
	height: 87px;
	top: 37px;
	left: 543px;
	background-color: green;
}
#subscribe-button {
	width: 90px;
	height: 55px;
	top: 1079px;
	left: 1265px;
	background-color: darkgreen;
}
```

</details>

<details>
	<summary>...should be main class with modificators</summary>

```css
.button {
	min-width: 90px;
	min-height: 55px;
	padding: 1em;
	background-color: green;
}
.button-dark {
	background-color: darkgreen;
}
```

</details>

Because of that I disagree with copy/pasting styles from spec, and instead of that I'm writing CSS that should look "as much as possible" the same as spec.

Later, when we start working on other pages and responsive, we'll have benefits like style reusability and stretching sizes.

ogonzal87 commented on src/components/Header/index.tsx:
```diff
@@ -0,0 +1,71 @@
+      </Link>
+      {content.navigationLinks.map(
+        (item, i): JSX.Element => (
+          <Link
```
We should use the Link from Gatsby to toggle the active state:
`<Link to="/learn" activeStyle={activeStyleTab} partiallyActive>`

ogonzal87 replied:
``` javascript
const activeStyleTab = {
  fontWeight: 'var(--font-weight-semibold)',
  color: 'var(--color-text-accent)',
  borderBottom: 'var(--space-04) inset var(--color-text-accent)',
};
```

ogonzal87 commented:
Hey @alexandrtovmach ! Thank you sooo much!! 

Totally agree. Not everything can be up to Spec. I made some comments on the new PR. 

Let me know if you have any questions.

alexandrtovmach commented:
@ogonzal87 Thank you and sorry for long response. I'll take a look in a few days

alexandrtovmach commented:
/gcbrun

bnb commented:
/gcbrun

alexandrtovmach commented:
@ogonzal87 @bnb Could you review it?

alexandrtovmach commented:
@nodejs/website-redesign

alexandrtovmach commented:
/gcbrun

alexandrtovmach commented:
Holidays end's up, and we need to unfreeze development process, please review this PR and let's go with the next steps. This PR is contains structural changes, that's going to be basis for the whole website, and I'm waiting for merge before implement other pages.

alexandrtovmach commented:
/gcbrun

SMotaal commented:
@alexandrtovmach do you want to join our hackathon this week — see #417

SMotaal commented:
/gcbrun

SMotaal commented:
@alexandrtovmach I tried to spotlight in today's biweekly meeting the critical compositor optimizations, and I personally expressed we want to at least try to separate out this and other critical contributions and land them even if in separate PRs.

I'd like to collaborate with you on that, more likely also other separate PR work we come to moving forward.

Let's get into this in this week's hackathon… Sounds good?

alexandrtovmach commented:
I've tried to slice parts of logic to separate PR, but it's too complicated, because everything in this PR connected:
- filestructure
- scss

We should merge this PR as it is or I'll close it
@nodejs/website-redesign please review and leave your opinion/vote (👍 merge / 👎 close). If we decide to merge it, I'll resolve conflicts and then I need a few approvals, if not I don't want to spend much time for nothing again =(

## Quick explanation why I think it should be landed:

### Filestructure changes:
Current [_root-level_ file structure](https://github.com/nodejs/nodejs.dev/tree/staging/src/components) with global CSS file is totally not okay, and it should be replaced with _domain structure_:
```
./Button
--/index.jsx
--/button.css
--/helpers/
----/iconsHandler.js
```
Other thing is for templates approach, which is Best Practice for Gatsby, and it's not used in right way in current `staging`.

### SCSS
We definitely should use some preprocessor to handle modular styles. In my PR I propose to use CSS modules, which works well with TS. Current staging structure "few global CSS files for everything" it's awful anti pattern that should be fixed as soon as possible.

alexandrtovmach commented:
We're not going to merge it, so it should be closed in favor to smaller PRs based on this
